rfcs/XXXX-gene-sets/text.md

```diff
@@ -0,0 +1,100 @@
+# Gene Sets Data Format
+
+## [Optional] Subtitle of my design 
+
+**Authors:** [Madison Dunitz](mailto:madison.dunitz@chanzuckerberg.com)
+
+**Approvers:** [Friendly Tech Person](mailto:some.nice.person@chanzuckerberg.com), [Friendly PM Person](mailto:some.nice.person@chanzuckerberg.com), [Friendly Tech Person2](mailto:some.nice.person@chanzuckerberg.com), [Girish Patangay](mailto:girish.patangay@chanzuckerberg.com), [Sara Gerber](mailto:sara.gerber@chanzuckerberg.com) 
+
+## tl;dr 
+
+This RFC contains the details about the data format for the gene set feature. It will support the export of differential
+ expression results and allow a user to upload their own gene sets for analysis. 
+ 
+## Glossery 
+- Gene set - List of genes
+- Differential expression - A statistic (typically logfold and p-value) describing the difference in gene expression between two sets of cells. It may refer to stats specific to one gene (differential expression value) or many (differential expression analysis). When referring to many genes it is typically capped (eg top 10 most differentially expressed genes)
+- LogFold - A ratio of the difference in expression levels. Typically calculated by taking the log2 of the gene expression value for each cell and then finding the mean of that value for each set of cells. Because the data is already log2 transformed you can then take the difference between the two means as the logfold value 
+## Problem Statement | Background
+
+Gene sets are the expected output of differential expression analysis in cellxgene. Two (non-overlaping) sets of cells 
```

Comment:
overlapping

Reply:
ok this is something I've been wondering about, what information do we get by comparing the difference in expression of the same cells?

Reply:
cc @colinmegill

Reply:
@MDunitz, @ambrosejcarr question regarding the science. We haven't had this constraint in the interface yet. Doing diffexp via categorical metadata, ie., `Tissue lung vs Tissue heart` or `FooCatgry SetA vs FooCatgry SetB` introduces the same constraint we currently have in categorical metadata — a cell can only have one category and could thus only be in one set.

Reply:
I am fairly confident that it is statistically invalid to include the same observation in both samples being compared by a differential expression test. I can't imagine a case where a user would want to do this, and I haven't tried. From Wikipedia: 

_The independent samples t-test is used when two separate sets of independent and identically distributed samples are obtained, one from each of the two populations being compared._

Reply:
@colinmegill @signechambers1 Do we need to make changes for this? 
Also Colin re categorical data -- unless we restrict diff exp comparisons to within a category I dont think this will solve the problem. What if a user wants to compare CategoryA.Label1 to CategoryB.Label1? Those cell sets could include the same cells.

Reply:
@MDunitz In moving diffexp to categories, users can only do diffexp comparison within the same category. This makes the lease sense, and creating a new user generated category, is the most cumbersome, if the task is comparing two labels in separate categories.

But it enables persistence of arbitrarily complex spatial and continuous selections.

Reply:
Here, the scientist wants to compare cells labeled Female and those labeled Heart. I created a new label Female first, since this is the more general category that I want to 'subtract' from. Then I created another label, Heart, and assigned cells to that — this operation _subtracts_ cells from Female because we don't allow overlaps / multiple assignments within a category. Cells labeled female are *reassigned* to Heart. 

This is an important caveat and edge case for doing diffexp via categorical metadata, where we benefit from fast precompute and persistence of spatial selections, but do not allow overlaps. The present edge case we are solving for, of course, is that users can do arbitrary diffexp but not persist it.

It should also be said that this feature is _not_ a blocker for gene sets — however we do diffexp, we can dump the results into a geneset.

![image](https://user-images.githubusercontent.com/1770265/97206049-9eaf1000-178e-11eb-990d-125abb52c9b2.png)